pkg/reconciler/route/route.go

```diff
@@ -480,10 +447,83 @@ func (c *Reconciler) ensureFinalizer(route *v1alpha1.Route) error {
 	return err
 }
 
+func (c *Reconciler) updateRouteStatusURL(ctx context.Context, route *v1alpha1.Route, clusterLocalServices sets.String) error {
+	mainRouteServiceName, err := domains.HostnameFromTemplate(ctx, route.Name, "")
+	if err != nil {
+		return err
+	}
+
+	mainRouteMeta := route.ObjectMeta.DeepCopy()
+	labels.SetVisibility(mainRouteMeta, clusterLocalServices.Has(mainRouteServiceName))
+
+	host, err := domains.DomainNameFromTemplate(ctx, *mainRouteMeta, route.Name)
+	if err != nil {
+		return err
+	}
+
+	route.Status.URL = &apis.URL{
+		Scheme: "http",
+		Host:   host,
+	}
+
+	return nil
+}
+
+func (c *Reconciler) getServiceNames(ctx context.Context, route *v1alpha1.Route) (serviceNames, error) {
+	// Populate existing service name sets
+	existingServices, err := c.getServices(route)
+	if err != nil {
+		return serviceNames{}, err
+	}
+	existingServiceNames := resources.GetNames(existingServices)
+	existingClusterLocalServices := resources.FilterService(existingServices, resources.IsClusterLocalService)
+	existingClusterLocalServiceNames := resources.GetNames(existingClusterLocalServices)
+	existingPublicServiceNames := existingServiceNames.Difference(existingClusterLocalServiceNames)
+
+	// Populate desired service name sets
+	desiredServiceNames, err := resources.GetDesiredServiceNames(ctx, route)
+	if err != nil {
+		return serviceNames{}, err
```

Comment:
yeah, this doesn't look nice. Switch to `*serviceNames` and return nil here.

Reply:
Done! :)